### PR TITLE
[table service] Fix StringUtf8Coder and add VarIntCoder

### PR DESCRIPTION
**Pull request by sijie**
Descriptions of the changes in this PR:

*Motivation*

The getSerializeLen() returned by StringUtf8Coder is different from the serialized buffer size.

*Changes*

- Fix StringUtf8Coder
- Add tests to coders
- Add a VarIntCoder

> ---
> In order to uphold a high standard for quality for code contributions, Apache BookKeeper runs various precommit
> checks for pull requests. A pull request can only be merged when it passes precommit checks. However running all
> the precommit checks can take a long time, some trivial changes don't need to run all the precommit checks. You
> can check following list to skip the tests that don't need to run for your pull request. Leave them unchecked if
> you are not sure, committers will help you:
>
> - [x] [skip bookkeeper-server bookie tests]: skip testing `org.apache.bookkeeper.bookie` in bookkeeper-server module.
> - [x] [skip bookkeeper-server client tests]: skip testing `org.apache.bookkeeper.client` in bookkeeper-server module.
> - [x] [skip bookkeeper-server replication tests]: skip testing `org.apache.bookkeeper.replication` in bookkeeper-server module.
> - [x] [skip bookkeeper-server tls tests]: skip testing `org.apache.bookkeeper.tls` in bookkeeper-server module.
> - [x] [skip bookkeeper-server remaining tests]: skip testing all other tests in bookkeeper-server module.
> - [ ] [skip integration tests]: skip docker based integration tests. if you make java code changes, you shouldn't skip integration tests.
> - [ ] [skip build java8]: skip build on java8. *ONLY* skip this when *ONLY* changing files under documentation under `site`.
> - [ ] [skip build java9]: skip build on java9. *ONLY* skip this when *ONLY* changing files under documentation under `site`.
> ---

> ---
> Be sure to do all of the following to help us incorporate your contribution
> quickly and easily:
>
> If this PR is a BookKeeper Proposal (BP):
>
> - [ ] Make sure the PR title is formatted like:
>     `<BP-#>: Description of bookkeeper proposal`
>     `e.g. BP-1: 64 bits ledger is support`
> - [ ] Attach the master issue link in the description of this PR.
> - [ ] Attach the google doc link if the BP is written in Google Doc.
>
> Otherwise:
> 
> - [ ] Make sure the PR title is formatted like:
>     `<Issue #>: Description of pull request`
>     `e.g. Issue 123: Description ...`
> - [ ] Make sure tests pass via `mvn clean apache-rat:check install spotbugs:check`.
> - [ ] Replace `<Issue #>` in the title with the actual Issue number.
> 
> ---

### CODE REVIEW

**Comment by sijie** [comment]
This PR includes changes from #1506 to allow it pass precommit jobs.

File: stream/common/src/test/java/org/apache/bookkeeper/common/coder/TestByteArrayCoder.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+    private static final ByteArrayCoder TEST_CODER = ByteArrayCoder.of();
+
+    private static final List<byte[]> TEST_VALUES = Arrays.asList(
+        new byte[]{0xa, 0xb, 0xc},
+        new byte[]{0xd, 0x3},
+        new byte[]{0xd, 0xe},
+        new byte[]{});
+
+    @Test(timeout = 10000)
```
Comment (eolivelli) [inline stream/common/src/test/java/org/apache/bookkeeper/common/coder/TestByteArrayCoder.java]: Are we using Timeout in these tests?

Comment (sijie) [reply stream/common/src/test/java/org/apache/bookkeeper/common/coder/TestByteArrayCoder.java]: these are code made a very long time ago, so there were still `@Test` annotations. will remove them.

Comment (sijie) [reply stream/common/src/test/java/org/apache/bookkeeper/common/coder/TestByteArrayCoder.java]: removed

**Comment by sijie** [comment]
run integration tests

**Comment by sijie** [comment]
run integration tests

**Pull request closed without merging**